### PR TITLE
Fix example tune-pytorch where the checkpoint path may be named differently

### PR DESCRIPTION
**Pull request by jingdong00**
## Why are these changes needed?
Following the Tune with Pytorch example (https://microsoft.github.io/FLAML/docs/Examples/Tune-PyTorch), the following line will give an error saying that the checkpoint has no attribute 'value'
checkpoint_path = os.path.join(best_trial.checkpoint.value, "checkpoint"). 
This is because that best_tria.checkpoint is a Trial object defined in ray.tune.experiment.trial . In the newer version of ray, the checkpoint value (the path) attribute is named 'dir_or_data'. See https://docs.ray.io/en/latest/_modules/ray/tune/experiment/trial.html for more info.

## Related issue number
#852 

## Checks

- [x ] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR, or I've made sure [lint with flake8](https://github.com/microsoft/FLAML/blob/816a82a1155b4de4705b21a615ccdff67c6da379/.github/workflows/python-package.yml#L54-L59) output is two 0s.
- [x ] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [ na] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x ] I've made sure all auto checks have passed.

### CODE REVIEW

File: flaml/automl/automl.py [hunk 1]
```diff
@@ -2566,6 +2566,11 @@ def cv_score_agg_func(val_loss_folds, log_metrics_folds):
         self._use_ray = use_ray or n_concurrent_trials > 1
         # use the following condition if we have an estimation of average_trial_time and average_trial_overhead
         # self._use_ray = use_ray or n_concurrent_trials > ( average_trail_time + average_trial_overhead) / (average_trial_time)
+
+        # If no time_budget and no max_iter is specified, then effectively zero-shot AutoML is used. 
+        if time_budget is -1 and max_iter is None:
+            logger.info('Neither time_budegt nor max_iter is specified, zero-shot ML is used. ')
+
```
Comment (sonichi) [inline flaml/automl/automl.py:2573]: Remove this change

File: website/docs/Examples/Tune-PyTorch.md [hunk 2]
```diff
@@ -283,4 +284,4 @@ Files already downloaded and verified
 Best trial test set accuracy: 0.6294
 ```
 
-[Link to notebook](https://github.com/microsoft/FLAML/blob/main/notebook/tune_pytorch.ipynb) | [Open in colab](https://colab.research.google.com/github/microsoft/FLAML/blob/main/notebook/tune_pytorch.ipynb)
\ No newline at end of file
+[Link to notebook](https://github.com/microsoft/FLAML/blob/main/notebook/tune_pytorch.ipynb) | [Open in colab](https://colab.research.google.com/github/microsoft/FLAML/blob/main/notebook/tune_pytorch.ipynb)
```
Comment (sonichi) [inline website/docs/Examples/Tune-PyTorch.md:287]: Update the notebook too.
Also, check whether [test_pytorch_cifar10.py](https://github.com/microsoft/FLAML/blob/main/test/tune/test_pytorch_cifar10.py) needs to be updated.

Comment (jingdong00) [reply website/docs/Examples/Tune-PyTorch.md:287]: I've updated the notebook and the test

**Review by sonichi — APPROVED** [review]
I suggest creating an issue to add a test of the notebook, like what's done in #846 .

**Review by skzhang1 — APPROVED** [review]

**Comment by skzhang1** [comment]
> I suggest creating an issue to add a test of the notebook, like what's done in #846 .

I merge this PR for the time being. Maybe we can open a new PR to support this test @jingdong00

**Comment by jingdong00** [comment]
> #846

sure. I'll open a new PR later for the test.

**Pull request merged**